File: 2461_Maximum_Sum_of_Distinct_Subarray_With_Length_K/initialSolution.java
import java.util.HashSet;

class Solution {
    public long maximumSubarraySum(int[] nums, int k) {
        long maxSum = 0;
        long currentSum = 0;
        HashSet<Integer> windowSet = new HashSet<>();

        int left = 0;
        for (int right = 0; right < nums.length; right++) {
            while (windowSet.contains(nums[right])) {
                windowSet.remove(nums[left]);
                currentSum -= nums[left];
                left++;
            }

            windowSet.add(nums[right]);
            currentSum += nums[right];

            if (right - left + 1 == k) {
                maxSum = Math.max(maxSum, currentSum);
                windowSet.remove(nums[left]);
                currentSum -= nums[left];
                left++;
            }
        }

        return maxSum;
    }
}

File: 2461_Maximum_Sum_of_Distinct_Subarray_With_Length_K/initialSolution.go
package main

func maximumSubarraySum(nums []int, k int) int64 {
	var maxSum int64 = 0
	var currentSum int64 = 0

	var left = 0

	var windowMap map[int]int = make(map[int]int)

	for right := 0; right < len(nums); right++ {

		currentElement := nums[right]
		currentSum += int64(currentElement)
		windowMap[currentElement]++

		for windowMap[nums[right]] > 1 {
			leftElement := nums[left]
			currentSum -= int64(leftElement)
			windowMap[leftElement]--
			left++

		}

		if right-left+1 == k {
			if currentSum > maxSum {
				maxSum = currentSum
			}
			leftElement := nums[left]
			currentSum -= int64(leftElement)
			windowMap[leftElement]--
			left++
		}

	}

	return maxSum

}
